src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -473,28 +473,27 @@ private static function mergeQueryString(?string $queryString, array $queryArray
         $query = [];
 
         if (null !== $queryString) {
-            foreach (explode('&', $queryString) as $v) {
-                if ('' !== $v) {
-                    $k = urldecode(explode('=', $v, 2)[0]);
-                    $query[$k] = (isset($query[$k]) ? $query[$k].'&' : '').$v;
-                }
-            }
+            parse_str($queryString, $query);
```

Comment:
we should not use parse_url, it replaces some characters and breaks legit query strings - this code should stay as is to me